pkg/kubecost/asset.go

```diff
@@ -3024,26 +3063,47 @@ type AssetSetRangeResponse struct {
 }
 
 func (asr *AssetSetRange) UTCOffset() time.Duration {
-	if asr.Length() == 0 {
+	if asr == nil {
 		return 0
 	}
 
-	as, err := asr.Get(0)
-	if err != nil {
+	asr.RLock()
+	defer asr.RUnlock()
+
+	if len(asr.assets) == 0 {
 		return 0
 	}
+
+	as := asr.assets[0]
+	if as == nil {
+		return 0
+	}
+
 	return as.UTCOffset()
 }
 
 // Window returns the full window that the AssetSetRange spans, from the
 // start of the first AssetSet to the end of the last one.
 func (asr *AssetSetRange) Window() Window {
-	if asr == nil || asr.Length() == 0 {
+	if asr == nil {
 		return NewWindow(nil, nil)
 	}
 
-	start := asr.assets[0].Start()
-	end := asr.assets[asr.Length()-1].End()
+	asr.RLock()
+	defer asr.RUnlock()
+
+	if len(asr.assets) == 0 {
+		return NewWindow(nil, nil)
+	}
+
+	startAs := asr.assets[0]
+	endAs := asr.assets[len(asr.assets)-1]
+	if startAs == nil || endAs == nil {
+		return NewWindow(nil, nil)
+	}
+
+	start := startAs.Start()
+	end := endAs.End()
```

Comment:
We were just YOLOing this before. Acquiring the read lock twice during each `Length()` call. Just need a single RLock here and direct access.